src/Umbraco.Web.UI.Client/src/views/common/overlays/mediaPicker/mediapicker.controller.js

```diff
@@ -15,15 +15,33 @@ angular.module("umbraco")
             $scope.multiPicker = (dialogOptions.multiPicker && dialogOptions.multiPicker !== "0") ? true : false;
             $scope.startNodeId = dialogOptions.startNodeId ? dialogOptions.startNodeId : -1;
             $scope.cropSize = dialogOptions.cropSize;
-            $scope.lastOpenedNode = localStorageService.get("umbLastOpenedMediaNodeId");
-            if ($scope.onlyImages) {
-                $scope.acceptedFileTypes = mediaHelper
-                    .formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.imageFileTypes);
-            } else {
-                $scope.acceptedFileTypes = !mediaHelper
-                    .formatFileTypes(Umbraco.Sys.ServerVariables.umbracoSettings.disallowedUploadFiles);
-            }
-            $scope.maxFileSize = Umbraco.Sys.ServerVariables.umbracoSettings.maxFileSize + "KB";
+            $scope.lastOpenedNode = localStorageService.get("umbLastOpenedMediaNodeId");
+
+            var umbracoSettings = Umbraco.Sys.ServerVariables.umbracoSettings;
+            var allowedUploadFiles = mediaHelper.formatFileTypes(umbracoSettings.allowedUploadFiles);
+            if ($scope.onlyImages) {
+                // If whitelist provided, use just images from that
+                if (allowedUploadFiles !== '') {
```

Comment:
I understand what you're doing here but this is fundamentally different to what is currently happening - the list of imageFileTypes (which admittedly has been hijacked from it's original usage) determined what is being shown in the list. This would be a breaking change in that behavior.
Furthermore I think it would be unexpected behavior.. if you had already uploaded a bunch of gif files but they are no longer allowed to be uploaded after a configuration change, I would not expect them to suddenly now appear in the list any more.
Maybe I'm missing something in your thinking?